### ISSUE TITLE
disallow index as key on v-for

### ISSUE THREAD

**Issue by caihai123** [vuejs/eslint-plugin-vue#1746]
**Please describe what the rule should do:**
很多同事为了方便每次都使用index作为v-for的key值，看起来很难受

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
<ul>
   <!-- warn -->
  <li v-for="(item, index) in list" :key="index"></li>
</ul>

<ul>
   <!-- ok -->
  <li v-for="item in list" :key="item.id"></li>
</ul>
```

**Comment by ota-meshi** [comment]
I can't read the language (Is it Chinese?) you are using. Can you rewrite in English?

Do you want a rule that rejects using `index` in `:key`? 
If so, I think that rule would be useful to the user. However, it makes sense to use the second argument if the right-hand side is an object. If the rule reports it, it can be a false positive.
I think it's probably difficult to avoid false positives and implement rule.

**Comment by caihai123** [comment]
Is it possible to use object in v-for like this
```vue
<template>
  <ul>
    <!-- ok -->
    <li v-for="(value, key) in object" :key="key"></li>
 </ul>
```
The second parameter must be named key

**Comment by ota-meshi** [comment]
I don't think it's good to decide whether to report by name.

**Comment by ota-meshi** [comment]
Please write the issue in English so that someone can easily work on the issue.
